controllers/operator/manager.go

```diff
@@ -205,12 +207,25 @@ func (m *ODLMOperator) GetSubscription(ctx context.Context, name, namespace stri
 	} else if !apierrors.IsNotFound(err) {
 		return nil, err
 	}
-	subPkgKey := types.NamespacedName{
-		Name:      packageName[0],
-		Namespace: namespace,
+
+	subList := &olmv1alpha1.SubscriptionList{}
+	if err := m.Client.List(ctx, subList, &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			"operators.coreos.com/" + packageName + "." + namespace: "",
+		}),
```

Comment:
We change from checking the label in the subscription instead of the subscription's name by `packageName`.

Is it because the subscription created via OLM, will always have those kind of `package` label. But the name of subscription is uncertain, it may vary since it could be decided by user?

Reply:
Yes, the label is added by OLM. I am wondering if we should check these labels or just iterate all the subscriptions in the namespace.

Reply:
I think the current method is more elegant.

If we iterate the subscriptions in that namespace, we still need to check the labels in them. So why don't we just `List` the subscriptions with those labels and count their numbers.